rmw_fastrtps_cpp/src/functions.cpp

```diff
@@ -1750,10 +1702,9 @@ extern "C"
             {
                 void *data = guard_conditions->guard_conditions[i];
                 GuardCondition *guard_condition = (GuardCondition*)data;
-                if(!guard_condition->getHasTriggered())
-                {
-                    guard_conditions->guard_conditions[i] = 0;
-                }
+                // Guard conditions persist (don't set to null)
+                // We call getHasTriggered to reset the trigger state.
```

Comment:
ok. I confess I don't really like the "setting the handles to null" pattern in rmw/rclcpp because it requires managing a lot of boilerplate in the Executor, but I'm not going to spend time trying to make up a better pattern right now because I have so many pull requests in flight. It's also a bit hard to motivate keeping the boilerplate to manage the null handles in Executor for debugging purposes when there isn't an existing use case for it--I can see how this would be useful if we didn't have a completely monolithic system where we can edit and rebuild all the packages, but right now if I wanted to debug things I would just go directly into the RMW implementations... Anyway, if you both think this is worth preserving, I will fix up this pull request (and all the other rmw implementations) accordingly.